ffi/src/engine_data.rs

```diff
@@ -93,3 +93,25 @@ fn get_raw_arrow_data_impl(data: Box<dyn EngineData>) -> DeltaResult<*mut ArrowF
     let ret_data = Box::new(ArrowFFIData { array, schema });
     Ok(Box::leak(ret_data))
 }
+
+#[cfg(feature = "default-engine")]
+#[no_mangle]
+pub unsafe extern "C" fn get_engine_data(
+    data: *mut ArrowFFIData,
+    engine: Handle<SharedExternEngine>,
+) -> ExternResult<Handle<ExclusiveEngineData>> {
+    get_engine_data_impl(data)
```

Comment:
nit: convert the pointer back to reference here so the impl doesn't has one less unsafe thing to deal with
```suggestion
    get_engine_data_impl(&*data)
```
Even better, just pass by reference directly `data: &ArrowFFIData` and the pointer->reference conversion is baked into the call itself.

Can also have the impl return `ExclusiveEngineData` so you don't need the `map` here.

Reply:
think I did what you suggested, see other comment